workspace/python/ds-pack/2.1.1-base-python-3.10.13/Dockerfile

```diff
@@ -0,0 +1,147 @@
+FROM python:3.10.13-slim-bookworm
+
+# Set shell to be bash and fail any pipe if any step in the pipe fails
+SHELL ["/bin/bash", "-o", "pipefail", "-c"]
+
+# install system packages
+RUN apt-get update \
+    # Installation requirements
+    && apt-get install -y --no-install-recommends \
+    apt-transport-https \
+    ca-certificates \
+    software-properties-common \
+    curl \
+    gnupg2 \
+    # Set up Docker repository
+    && mkdir -p /etc/apt/keyrings \
+    && curl -fsSL https://download.docker.com/linux/debian/gpg | gpg --dearmor -o /etc/apt/keyrings/docker.gpg \
+    && echo \
+        "deb [arch=$(dpkg --print-architecture) signed-by=/etc/apt/keyrings/docker.gpg] https://download.docker.com/linux/debian \
+        $(lsb_release -cs) stable" | tee /etc/apt/sources.list.d/docker.list > /dev/null \
+    && apt-key adv --keyserver keyserver.ubuntu.com --recv-key '95C0FAF38DB3CCAD0C080A7BDC78B2DDEABC47B7' \
+    && add-apt-repository 'deb http://cloud.r-project.org/bin/linux/debian bookworm-cran40/' \
+    # node
+    && curl -sL https://deb.nodesource.com/setup_lts.x | bash - \
+    && apt-get update \
+    && apt-get install -y --no-install-recommends \
+    # install system packages
+    # jq is needed for the Peak Platform
+    jq \
+    nano \
+    vim \
+    fish \
+    zsh \
+    r-base=4.2.2.20221110-2 \
+    r-recommended=4.2.2.20221110-2 \
+    build-essential \
+    sudo \
+    nodejs \
+    git \
+    htop \
+    pandoc \
+    texlive \
+    texlive-science \
+    texlive-latex-extra \
+    texlive-luatex \
+    texlive-xetex \
+    texlive-pictures \
+    latexmk \
+    lmodern \
+    docker-ce \
+    less \
+    libxext6 \
+    libxrender1 \
+    libxtst6 \
+    libfreetype6 \
+    libxi6 \
+    openssl \
+    unzip \
+    libpq-dev \
+    libssl-dev \
+    libffi-dev \
+    && rm -rf /var/lib/apt/lists/*
+
+# Install AWS CLI v2
+RUN curl -fsSL "https://awscli.amazonaws.com/awscli-exe-linux-x86_64-2.7.4.zip" -o "awscliv2.zip" \
+    && unzip -q awscliv2.zip \
+    && ./aws/install \
+    && rm -rf aws awscliv2.zip
+
+# Install pyenv globally
+RUN curl https://pyenv.run | bash \
+    && mkdir -p /usr/local/pyenv \
+    && mv /root/.pyenv/* /usr/local/pyenv \
+    && rm -rf /root/.pyenv \
+    && echo 'export PYENV_ROOT="/usr/local/pyenv"' >> /etc/profile.d/pyenv.sh \
+    && echo 'export PATH="/usr/local/pyenv/bin:$PATH"' >> /etc/profile.d/pyenv.sh \
+    && echo 'eval "$(pyenv init --path)"' >> /etc/profile.d/pyenv.sh \
+    && echo 'eval "$(pyenv virtualenv-init -)"' >> /etc/profile.d/pyenv.sh \
```

Comment:
This is just to initialize pyenv's virtual environment management system by setting up the necessary functions and environment variables. It will not create any venv environment by default.

Reply:
Ahh, okay 👍🏽